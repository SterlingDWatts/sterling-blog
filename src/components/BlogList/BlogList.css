.BlogList--loading-splash {
  text-align: center;
  font-size: 24px;
  margin: 16px;
}

.BlogList--color-splash {
  color: #0060c7;
}

.BlogList--loading-splash svg {
  color: #fa813d;
}
