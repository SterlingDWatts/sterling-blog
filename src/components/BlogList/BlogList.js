import React, { Component } from "react";
import BlogListItem from "../BlogListItem/BlogListItem";
import BlogListContext from "../../contexts/BlogListContext";
import "./BlogList.css";
import BlogApiService from "../../services/blog-api-service";

class BlogList extends Component {
  static contextType = BlogListContext;

  static defaultProps = {
    recent: false,
    blogId: -1
  };

  componentDidMount() {
    this.context.clearError();
    BlogApiService.getBlogs()
      .then(this.context.setBlogList)
      .catch(this.context.setError);
  }

  renderBlogItems = blogs => {
    let blogList = blogs.filter(blog => blog.id !== Number(this.props.blogId));
    blogList = this.props.recent ? blogList.slice(0, 2) : blogList;
    return blogList.map(blog => {
      return <BlogListItem key={blog.id} {...blog} />;
    });
  };

  render() {
    const blogs = this.context.blogList;
    return <ul className="BlogList">{blogs && this.renderBlogItems(blogs)}</ul>;
  }
}

export default BlogList;
