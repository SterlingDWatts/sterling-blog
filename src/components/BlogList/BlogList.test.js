import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer";
import BlogList from "./BlogList";

describe("BlogList Component", () => {
  const blogs = [{
    id: 1,
    title: "test title",
    content: "test content",
    author: "test author",
    longPic: "https://picsum.photos/seed/three/900/510",
    squarePic: "https://picsum.photos/seed/three/255/255",
    views: 8,
    date: "December 22, 2019"
  }];
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter><BlogList blogs={blogs} /></BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div)
  })

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter><BlogList blogs={blogs} /></BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders the UI as expected with no blogs", () => {
    const emptyList = []
    const tree = renderer
      .create(
        <BrowserRouter><BlogList blogs={emptyList} /></BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
