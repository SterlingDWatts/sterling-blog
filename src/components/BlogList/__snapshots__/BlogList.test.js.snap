// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList Component renders the UI as expected 1`] = `
<ul
  className="BlogList"
/>
`;

exports[`BlogList Component renders the UI as expected with no blogs 1`] = `
<ul
  className="BlogList"
/>
`;
